io/plugins/eu.esdihumboldt.hale.io.gml.test/src/eu/esdihumboldt/hale/io/gml/writer/internal/StreamGmlWriter2Test.groovy

```diff
@@ -381,4 +381,50 @@ class StreamGmlWriter2Test {
 
 		result
 	}
+
+	@Test
+	void testNumbers() throws Exception {
+		testNumber("integerNumber", 12345)
+		testNumber("doubleNumber", 12345.123456789)
+		testNumber("floatNumber", 12345.6789F)
+		testNumber("longNumber", 1234567890123456789L)
+		testNumber("shortNumber", 12345 as short)
+		testNumber("decimalNumber", new BigDecimal("1234567890.123456789"))
+	}
+
+	void testNumber(String elementName, number) throws Exception {
+		// Load the target schema
+		def schema = loadSchema(getClass().getResource("/data/numbers/numbers.xsd").toURI())
+
+		// Create the instance
+		def instance = new InstanceBuilder(types: schema).PrimitiveTestType {
+			"$elementName"(number)
+		}
+
+		// Write file, validate and load it
+		def writer = new GmlInstanceWriter()
+		def xmlFile = writeFile(instance, schema, writer).getAbsolutePath()
+		def xml = new XmlSlurper().parse(xmlFile)
```

Comment:
The temporary file should be deleted when it is no longer needed, ideally also controlled by `DEL_TEMP_FILES` as in similar logic in this class.